resources/shaders/glbspshader.vs

```diff
@@ -0,0 +1,32 @@
+#version 410 core
+
+layout (location = 0) in vec3 vaPos;
+layout (location = 1) in vec2 vaTexUV;
+layout (location = 2) in vec2 vaTexLayer;
+layout (location = 3) in vec3 vaNormal;
+layout (location = 4) in float vaAttrib;
+
+out vec4 vertexColour;
+out vec2 texuv;
+flat out vec2 olayer;
+out vec3 vsPos;
+out vec3 vsNorm;
+out int vsAttrib;
```

Comment:
should be "flat out int vsAttrib;" or it will error out with: vertex shader output `vsAttrib' specifies no interpolation qualifier, but fragment shader input specifies flat interpolation qualifier